internal/cmd/oohelperd/http.go

```diff
@@ -102,13 +106,56 @@ func httpDo(ctx context.Context, config *httpConfig) {
 	reader := &io.LimitedReader{R: resp.Body, N: config.MaxAcceptableBody}
 	data, err := netxlite.ReadAllContext(ctx, reader)
 	ol.Stop(err)
+
+	h3Endpoint := ""
+	if !config.h3 {
+		h3Endpoint = getHTTP3Altsvc(resp)
+	}
+
 	config.Out <- ctrlHTTPResponse{
-		BodyLength: int64(len(data)),
-		Failure:    httpMapFailure(err),
-		StatusCode: int64(resp.StatusCode),
-		Headers:    headers,
-		Title:      measurexlite.WebGetTitle(string(data)),
+		BodyLength:   int64(len(data)),
+		DiscoveredH3: h3Endpoint,
+		Failure:      httpMapFailure(err),
+		StatusCode:   int64(resp.StatusCode),
+		Headers:      headers,
+		Title:        measurexlite.WebGetTitle(string(data)),
+	}
+}
+
+func getHTTP3Altsvc(resp *http.Response) string {
+	var responses []*http.Response
+	var request *http.Request
+
+	// go through chain of HTTP Request - Responses to find first Response
+	// TODO (kelmenhorst): this will not work for http:// URLs
```

Comment:
With this code we are searching for the very first HTTP response in the redirect chain (because we want to check whether the original domain supports h3). I found out that many/some/? hosts do not advertise h3 via Alt-Svc on an HTTP/1.1 response. Thus, measuring `http://cloudflare.com` will not cause a h3 follow-up, but `https://cloudflare.com` will.

Reply:
Ah, damn! I think perhaps we should document this as a limitation for the time being?

Reply:
Yes, sounds good!